### PR TITLE
Workflow action version updates

### PR DESCRIPTION
**Pull request by cyb3rko**
Bumping the actions to much newer versions

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,9 +7,10 @@ jobs:
     name: Check
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
         with:
+          distribution: 'zulu'
```
Comment (jmattheis) [inline .github/workflows/main.yml:13]: Why zulu and not temurin?

Comment (cyb3rko) [reply .github/workflows/main.yml:13]: We are using the same one as before:  
https://github.com/actions/setup-java/blob/main/docs/switching-to-v2.md

You can change it if you want:  
https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions

Comment (jmattheis) [reply .github/workflows/main.yml:13]: Ahh okay, do you have a preference? I think temurin is more open, as it's maintained  eclipse?

Comment (cyb3rko) [reply .github/workflows/main.yml:13]: Based on https://whichjdk.com/, temurin sounds good

Comment (cyb3rko) [reply .github/workflows/main.yml:13]: I mostly use Zulu or SapMachine, but only because they just work. I never bothered to change it for my projects.

Comment (jmattheis) [reply .github/workflows/main.yml]: Okay, then let's go with temurin, if the build succeeds (:.

**Review by jmattheis — APPROVED** [review]

**Pull request merged**